Homepage carousel with database implemented

Tobias-hubs opened the pull request:
**Description**
Implemented and updated the carousel component with movie data from MongoDB Atlas database. 

**Features**

- Installed Swiper to handle the UI of the carousel. Swipe with mouse or with buttons. 

- Using MUI for styling

- Added logic for fetching data from database, pictures for all movies and a banner in the bottom that shows Title, genre and length of the specific movie. 

- Added if and else state as well as defining a boolean variable to quickly change if only the three orignal movies is shown as we had in our previous project or if all movies should be shown. 

- Added dynamic loop to Swiper element. 

- Refactored Button logic to separate file.

- Added autoplay every 3sec and made buttons less prominent 

**Testing**
Manual testing at the moment.
Some bugs like wrong positioned elements in carousel component and buttons not worked were resolved. 


**Preview**
![image](https://github.com/user-attachments/assets/272dbdc9-b0da-429f-8833-12032153ebbe)

Jorlindstrom reviewed (approved):
Stylish with full width and styling of the film's title and length. Impressive that everything is retrieved from the database. good job! It would be nice if it was possible to make the slider change images by itself. I think the arrows feel too prominent

Tobias-hubs commented:
> Stylish with full width and styling of the film's title and length. Impressive that everything is retrieved from the database. good job! It would be nice if it was possible to make the slider change images by itself. I think the arrows feel too prominent

Thanks for great feedback!
Made a new commit adressing the requested changes. Made the buttons less prominent and added autoplay logic to change picture every 3 sec. Now its much more intuitive.

Tobias-hubs commented:
> **Description** Implemented and updated the carousel component with movie data from MongoDB Atlas database.
> 
> **Features**
> 
> * Installed Swiper to handle the UI of the carousel. Swipe with mouse or with buttons.
> * Using MUI for styling
> * Added logic for fetching data from database, pictures for all movies and a banner in the bottom that shows Title, genre and length of the specific movie.
> * Added if and else state as well as defining a boolean variable to quickly change if only the three orignal movies is shown as we had in our previous project or if all movies should be shown.
> * Added dynamic loop to Swiper element.
> * Refactored Button logic to separate file.
> 
> **Testing** Manual testing at the moment. Some bugs like wrong positioned elements in carousel component and buttons not worked were resolved.
> 
> **Preview** ![image](https://private-user-images.githubusercontent.com/180289090/444060286-272dbdc9-b0da-429f-8833-12032153ebbe.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDczMTA3NTksIm5iZiI6MTc0NzMxMDQ1OSwicGF0aCI6Ii8xODAyODkwOTAvNDQ0MDYwMjg2LTI3MmRiZGM5LWIwZGEtNDI5Zi04ODMzLTEyMDMyMTUzZWJiZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNTE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDUxNVQxMjAwNTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00ZTc3MGQ1N2MwMzlhYTJhZTAyYWY4NDM3MmNjYzk3Mjc4ZmJjMzQ5YWQyODUzOTU0MmUwZGM0Y2UyMWM0OTVlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.jWD1Oe8iRwmCuS4dP-2VaaIkWhwQ1ZEcwIQ8pY9Nu88)

**Changes**
Added autoplay logic and made the navigation buttons less prominent.

Bekhalm reviewed (approved):
Looks really good, nice job! Just like the design.  OK to merge!
